### PR TITLE
Imagenet in 18 minutes entries for 4 and 8 machines

### PR DESCRIPTION
**Pull request by bearpelican**
Using similar schedule and training tricks as this PR:
https://github.com/stanford-futuredata/dawn-bench-entries/pull/53

### CODE REVIEW

**Comment by codyaustun** [comment]
@bearpelican please fix the failed checks. You need to change the column headers in the TSV files from `top1` and `top5` to `top1Accuracy` and `top5Accuracy` respectively.

File: ImageNet/train/dawn_4_machine_30min.json [hunk 1]
```diff
@@ -0,0 +1,43 @@
+{
+    "version": "v1.0",
+    "author": "Andrew Shaw, Yaroslav Bulatov, Jeremy Howard",
+    "authorEmail": "andrew.t.shaw@gmail.com",
+    "framework": "ncluster / pytorch",
+    "codeURL": "https://github.com/diux-dev/imagenet18/blob/master/training/train_imagenet_nv.py",
+    "model": "Resnet 50",
+    "hardware": "32 * V100 (4 machines - AWS p3.16xlarge)",
+    "costPerHour": 24.48,
```
Comment (codyaustun) [inline ImageNet/train/dawn_4_machine_30min.json]: I think this is the wrong value for `costPerHour`. You used 4 p3.16xlarge machines, so shouldn't the cost be 97.92 (4x24.48)?

File: ImageNet/train/dawn_8_machine_19min.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
+{
+    "version": "v1.0",
+    "author": "Andrew Shaw, Yaroslav Bulatov, Jeremy Howard",
+    "authorEmail": "andrew.t.shaw@gmail.com",
+    "framework": "ncluster / pytorch",
+    "codeURL": "https://github.com/diux-dev/imagenet18/blob/master/training/train_imagenet_nv.py",
+    "model": "Resnet 50",
+    "hardware": "64 * V100 (8 machines - AWS p3.16xlarge)",
+    "costPerHour": 24.48,
```
Comment (codyaustun) [inline ImageNet/train/dawn_8_machine_19min.json]: I think this is the wrong value for `costPerHour`. You used 8 p3.16xlarge machines, so shouldn't the cost be 195.84 (8x24.48)?

File: ImageNet/train/dawn_8_machine_19min.json [hunk 3]
```diff
@@ -0,0 +1,44 @@
+{
+    "version": "v1.0",
+    "author": "Andrew Shaw, Yaroslav Bulatov, Jeremy Howard",
+    "authorEmail": "andrew.t.shaw@gmail.com",
+    "framework": "ncluster / pytorch",
```
Comment (codyaustun) [inline ImageNet/train/dawn_8_machine_19min.json]: Please add the PyTorch version number

File: ImageNet/train/dawn_8_machine_19min.json [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    "codeURL": "https://github.com/diux-dev/imagenet18/blob/master/training/train_imagenet_nv.py",
```
Comment (codyaustun) [inline ImageNet/train/dawn_8_machine_19min.json]: Please include a commit hash. You can either update the `codeURL` field or add a separate `commitHash` field.

File: ImageNet/train/dawn_4_machine_30min.json [hunk 5]
```diff
@@ -0,0 +1,43 @@
+{
+    "version": "v1.0",
+    "author": "Andrew Shaw, Yaroslav Bulatov, Jeremy Howard",
+    "authorEmail": "andrew.t.shaw@gmail.com",
+    "framework": "ncluster / pytorch",
+    "codeURL": "https://github.com/diux-dev/imagenet18/blob/master/training/train_imagenet_nv.py",
```
Comment (codyaustun) [inline ImageNet/train/dawn_4_machine_30min.json]: Please include a commit hash. You can either update the `codeURL` field or add a separate `commitHash` field.

File: ImageNet/train/dawn_4_machine_30min.json [hunk 6]
```diff
@@ -0,0 +1,43 @@
+{
+    "version": "v1.0",
+    "author": "Andrew Shaw, Yaroslav Bulatov, Jeremy Howard",
+    "authorEmail": "andrew.t.shaw@gmail.com",
+    "framework": "ncluster / pytorch",
```
Comment (codyaustun) [inline ImageNet/train/dawn_4_machine_30min.json]: Please add the PyTorch version number

**Review by codyaustun — CHANGES_REQUESTED** [review]

**Comment by codyaustun** [comment]
Can you change the filenames? They shouldn't start with dawn. Here are the instructions from the [README.md](https://github.com/stanford-futuredata/dawn-bench-entries#imagenet-training)

> JSON and TSV files are named `[author name]_[model name]_[hardware tag]_[framework].json`, similar to
`dawn_resnet56_1k80-gc_tensorflow.[json|tsv]`. Put the JSON and TSV files in the `ImageNet/train/` sub-directory.

File: ImageNet/train/dawn_8_machine_19min.json (hunk 2 above)
Comment (bearpelican) [reply ImageNet/train/dawn_8_machine_19min.json]: Oops good catch!

**Comment by bearpelican** [comment]
07ec7d0 should have all the updates!

File: ImageNet/train/fastai_resnet50_p3_4_machine_pytorch.json [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+            {"image_size": 288, "epochs": 2}
+        ],
+        "schedule": [
+          {"learning_rate": [1,2], "epochs": 6, "linear": true},
+          {"learning_rate": 2, "epochs": 5},
+          {"learning_rate": [4,2], "epochs": 2, "linear": true},
+          {"learning_rate": 1.75, "epochs": 3},
+          {"learning_rate": [1.75,0.175], "epochs": 7, "linear": true},
+          {"learning_rate": [0.175,0.0175], "epochs": 4, "linear": true},
+          {"learning_rate": [0.01,0.001], "epochs": 2, "linear": true}
```
Comment (codyaustun) [inline ImageNet/train/fastai_resnet50_p3_4_machine_pytorch.json]: The epochs don't add up to the same number here and don't match the TSV. Is there something missing? `schedule` has 29 epochs, `imageSize` has 30, `batchSize` has 30, and the TSV has 30 lines.

File: ImageNet/train/fastai_resnet50_p3_8_machine_pytorch.json [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+            {"image_size": 288, "epochs": 6}
+        ],
+        "schedule": [
+          {"learning_rate": [1.92,3.84], "epochs": 6, "linear": true},
+          {"learning_rate": 3.84, "epochs": 5},
+          {"learning_rate": [3.84,1.92], "epochs": 3, "linear": true},
+          {"learning_rate": 1.92, "epochs": 3},
+          {"learning_rate": [1.92,0.336], "epochs": 6, "linear": true},
+          {"learning_rate": [0.336,0.0336], "epochs": 6, "linear": true},
+          {"learning_rate": [0.0192,0.00192], "epochs": 6, "linear": true}
```
Comment (codyaustun) [inline ImageNet/train/fastai_resnet50_p3_8_machine_pytorch.json]: `batchSize`, `imageSize`, and `schedule` all match with 35 epochs, but there are 36 rows in the TSV. What is the reason for the difference?

**Review by codyaustun — CHANGES_REQUESTED** [review]
Thanks for addressing all of the feedback so far. I just want to make sure we have all the details for the learning rate schedule.

**Comment by bearpelican** [comment]
4 machine was an off by one error - 52481c9. Math is hard for me

8 machine should be fixed. I forgot that run was from an older schedule (which had useless epochs at the end)

**Review by codyaustun — APPROVED** [review]

**Comment by codyaustun** [comment]
Great! Everything looks good. I'll merge everything in by tomorrow evening
